target/scripts/helper-functions.sh

```diff
@@ -225,33 +225,33 @@ export -f _notify
 CHKSUM_FILE=/tmp/docker-mailserver-config-chksum
 
 # Compute checksums of monitored files,
-# returned output is lines of hashed content + filepath pairs.
+# returned output on `stdout`: hash + filepath tuple on each line
 function _monitored_files_checksums
 {
   # If a wildcard path pattern (or an empty ENV) would yield an invalid path
   # or no results, `shopt -s nullglob` prevents it from being added.
   shopt -s nullglob
+  local CERT_FILES
 
   # React to any cert changes within the following letsencrypt locations:
-  local CERT_FILES=(
+  CERT_FILES=(
     /etc/letsencrypt/live/"${SSL_DOMAIN}"/*.pem
     /etc/letsencrypt/live/"${HOSTNAME}"/*.pem
     /etc/letsencrypt/live/"${DOMAINNAME}"/*.pem
   )
 
-  # CERT_FILES should expand to separate paths, not a single string;
-  # otherwise fails to generate checksums for these file paths.
-  #shellcheck disable=SC2068
-  (
-    cd /tmp/docker-mailserver || exit 1
-    exec sha512sum 2>/dev/null -- \
-      postfix-accounts.cf \
-      postfix-virtual.cf \
-      postfix-aliases.cf \
-      dovecot-quotas.cf \
-      /etc/letsencrypt/acme.json \
-      ${CERT_FILES[@]}
-  )
+  if [[ ! -d /tmp/docker-mailserver ]]
+  then
+    return 1
+  fi
+
+  sha512sum 2>/dev/null --                     \
+    /tmp/docker-mailserver/postfix-accounts.cf \
+    /tmp/docker-mailserver/postfix-virtual.cf  \
+    /tmp/docker-mailserver/postfix-aliases.cf  \
+    /tmp/docker-mailserver/dovecot-quotas.cf   \
+    /etc/letsencrypt/acme.json                 \
+    "${CERT_FILES[@]}"
```

Comment:
I think, the quotes must be removed. See https://github.com/docker-mailserver/docker-mailserver/pull/2326

```suggestion
    ${CERT_FILES[@]}
```

Reply:
Update: Now I think the quotes are fine. Initially, no array `[@]` was used, but just `${CERT_FILES}`. Quoting this, leads to a problem. Hower using a quoted array like above should be fine.

@polarathene Agree?

Reply:
I'd like to wait until I see what tests are saying because right now, I do not see why re-splitting of elements is needed here. Because of

> `# CERT_FILES should expand to separate paths, not a single string;`

it look like to me that there was a misconception that the array is expanded but is a single string, which is not the case. I could be wrong though - let's see.

Reply:
See my comment above 😉

Reply:
> See my comment above 😉 

You beat me to it 🙈

Reply:
Confirmed - this is not an issues :D

Reply:
I think previously it was a for loop that appended to the `CERT_FILES` variable, and the quoted string was preventing the paths being treated separately for input... then I refactored to an array but didn't know if it'd behave different so stuck with what worked? :sweat_smile: 

I trust you two since the you both know shell-fu better than me and the tests I wrote are still passing :+1: